crates/flowctl/src/local_specs.rs

```diff
@@ -119,11 +119,11 @@ async fn validate(
             .as_secs();
 
         let db_path = std::env::temp_dir().join(format!("flowctl_{seconds}.sqlite"));
-        build::persist(Default::default(), &db_path, Ok(&out)).expect("failed to write build DB");
+        build::persist(Default::default(), &db_path, &out).expect("failed to write build DB");
         tracing::debug!(db_path=%db_path.to_string_lossy(), "wrote debugging database");
     }
 
-    out
+    out.into_parts()
```

Comment:
I'm thinking that in the future it seems likely we may want to refactor the flowctl code paths to also use `BuildOutput` directly. But for now, I left it using the old tuple in order to keep the initial scope of changes small and quicker to review and test.